### PR TITLE
Add sample weights to DataFrameIterator

### PR DESCRIPTION
**Pull request by rragundez**
### Summary
Currently there is no implementation of sample weights in DataFrameIterator or flow_from_dataframe. This PR adds that functionality. 
For example with this PR is now possible to train a model using sample weights, given that a column of weights is added to the pandas dataframe.
```
generator = ImageDataGenerator().flow_from_dataframe(
    df,
    weight_col='weights'
)
model.fit_generator(generator, epochs=2, steps_per_epoch=10)
```
### Related Issues
Solves #147 

### PR Overview

- [y ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ y] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ y] This PR is backwards compatible [y/n]
- [ n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras_preprocessing/image/directory_iterator.py [hunk 1]
```diff
@@ -147,3 +147,8 @@ def filepaths(self):
     @property
     def labels(self):
         return self.classes
+
+    @property  # mixin needs this property to work
+    def sample_weight(self):
+        # no sample weights will be returned
+        return None
```
Comment (Dref360) [inline keras_preprocessing/image/directory_iterator.py]: Suggestion: Wouldn't returning an array of ones be cleaner?

Comment (rragundez) [reply keras_preprocessing/image/directory_iterator.py]: you might be right. Let me take a look how `NumpyArrayIterator` does it, but I think it does something similar, as in not returning any weights at all, hence saving some computation when calculating the loss function down the line.

**Comment by hbarovertwo** [comment]
@rragundez How can this work when doing multi-label classification? Many possible labels and each image can have at least one label. Elements in my labels column in dataframe look like these tuples (lbl1, lbl2, lbl3,...) so corresponding weight elements should look like (weight_lbl1, weight_lbl2,...)?

**Comment by rragundez** [comment]
@hbarovertwo unfortunately you cannot, hence the name sample weights. In this blog I make a similar example https://blog.godatadriven.com/rod-keras-multi-label
basically what I state about class imbalances in multi-label settings

"So what can we do about it?... pray. I won't go into detail but one way to counter the imbalances is with a combination of class weights and sample weights."

Trying to be respectful and honest, I'm also almost sure that that your weights (weight_lbl1, weight_lbl2,...) don't do what you think they do. In order to have balance it has to be across observations and labels, which sometimes has no solution (making one balance better makes the other worst most of the time). The best approach depends on the specifics but I would make sample weights and class weights that give me a good approximation to both balance. Good luck! let me know if you found a nice way to do it.

**Comment by hbarovertwo** [comment]
I was hoping to calculate batch weights on the fly but it didn't work out for me. I may revisit the problem later.
